### ISSUE TITLE
card-header-title: Inline html elements "swallow" adjoining whitespace.

### ISSUE THREAD

**Issue by rkofman** [jgthms/bulma#3650]
This is about Bulma. I believe this is a bug, but I may incorrect or it may be easier to fix as documentation.

### Overview of the problem

This is about the Bulma **CSS framework**
I'm using Bulma **version** [0.9.4]
My **browser** is: Google Chrome Version `113.0.5672.92 (Official Build) (arm64)`
I am sure this issue is **not a duplicate**

### Description

When inline elements are used as direct children of a `card-header-title`, they will disrupt the inline flow of text and erase any adjoining whitespace. There are workaround, but if the behavior is expected, it could use better documentation.

<img width="211" alt="Screenshot 2023-05-22 at 2 31 27 AM" src="https://github.com/jgthms/bulma/assets/35267/435aae7d-2219-4ec7-9c0b-272ff1e81d4c">

### Steps to Reproduce

1. Have a card with a title that includes html similar to the following:
```
<h3 class="card-header-title">Item <mark>match</mark>ing search</h3>
```
2. For example, see this [jsfiddle](https://jsfiddle.net/xwq92Lun/1/)
3. Observe the resulting whitespace

### Expected behavior

There should be a single space between `Item` and `matching`:

<img width="231" alt="Screenshot 2023-05-22 at 2 31 38 AM" src="https://github.com/jgthms/bulma/assets/35267/367cf2dc-f962-4258-85cf-54fdd1595f13">

### Actual behavior

The word "Item" runs right into "matching" with no whitespace:

<img width="211" alt="Screenshot 2023-05-22 at 2 31 27 AM" src="https://github.com/jgthms/bulma/assets/35267/435aae7d-2219-4ec7-9c0b-272ff1e81d4c">

### Workarounds
1. One workaround is to wrap the contents of `card-header-title` in a separate element -- so that its children do not automatically become flex items. This causes extra markup, and is not mentioned anywhere in the documentation as a necessary step:
```
<div class="card-header-title"><h3>Item <mark>match</mark>ing search</h3></div>
```
2. Another workaround is to manually apply a `display: block` to `card-header-title`. e.g:
```
<h3 class="card-header-title" style="display:block;">Item <mark>match</mark>ing search</h3>
```
### Thoughts on causes (& potential fixes)

1. This behavior occurs because of `display:flex` on `card-header title` ([link to code](https://github.com/jgthms/bulma/blob/d2d3c3dd7f2e4076b84e63efc998252851913a72/sass/components/card.sass#L49)). One way to fix it would be to simply remove that line (and also `align-items: center` which would no longer be valid. However, this would break the ability to add `.is-centered` as mentioned in the [`0.5.3` changelog](https://github.com/jgthms/bulma/blob/d2d3c3dd7f2e4076b84e63efc998252851913a72/CHANGELOG.md?plain=1#L1085).
2. If the expectation is that `card-header-title` is a _container for other elements that contain text_ and is not expected to contain text directly, the documentation should be updated to reflect this expectation. For example, the [Card component docs](https://github.com/jgthms/bulma/blob/d2d3c3dd7f2e4076b84e63efc998252851913a72/docs/cyp/components/card.html#L8) could replace their `<p class="card-header-title">Card header</p>` example with `<div class="card-header-title"><p>Card header</p></div>`

I am happy to submit a PR for either approach, but would like guidance (especially since solution # 1 would break existing functionality and need a bit more work to be less disruptive).

**Comment by dh-i** [comment]
@rkofman You are correct with the point 1 and guess it is the expected behaviour, that is at the best left as it is.

For the issue you are facing you can use HTML Entity (&nbsp -> non-breaking space) as below

```html
<h3 class="card-header-title">Item&nbsp<mark>match</mark>ing search</h3>
```
